packages/inter-protocol/src/stakeFactory/stakeFactoryKit.js

```diff
@@ -370,17 +367,56 @@ const potBehavior = {
     const { zcf } = state;
     const { helper } = facets;
     assert(state.open);
+
+    // Or should this be more like the AdjustBalancesProposalShape in
+    // vaultFactory, that allows both records to have both properties?
```

Comment:
I don't think so... but only because I don't see a case where you would want to give both collateral and debt-brand, or want both. Why is that allowed in vaultFactory?

Reply:
The requirement for VaultFactory AdjustBalances is that you can "give" or "want" without specifying the other. 

I don't know either of a good use case for giving the same brands as you're wanting, but there's also not a strong reason to forbid that and allowing it makes the shape simpler to implement the above requirement.

Doesn't the StakeFactory have the same requirement? i.e. that you can specify one and the other is implied by your staked holdings

Reply:
I can't tell whether this is redundant with what @turadg said, but I want to be explicit:

with VF AdjustBalances, you can give both or want both or give one and want the other. I don't think it ever makes sense to have the same keyword in both give and want.

Reply:
> If so we lack test coverage for that case ...

That's where my mind went too.

I'm not sure I can justify the time to swap in the relevant context for this just now.

Reply:
The goal for this PR is to express with patterns at least the restrictions that were expressed by `assertProposalShape` and similar, but still be unrestrictive enough to allow everything we should be allowing. So, based on what I think I understood of the above comments, I did revise this pattern to be like the `AdjustBalancesProposalShape` from VF, that is more accepting.

PTAL and let me know if I misunderstood. Leaving this Conversation unresolved until I hear back.